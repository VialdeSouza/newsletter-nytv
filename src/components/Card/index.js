import PropTypes from 'prop-types';
import * as React from 'react';
import {TouchableOpacity} from 'react-native';
import {Card, Title, Paragraph} from 'react-native-paper';
import Icon from 'react-native-vector-icons/MaterialIcons';
import useStyles from './styles';

const MainCard = ({title, description, url, onPress}) => {
  const styles = useStyles();

  return (
    <Card style={styles.item}>
      <Card.Cover source={{uri: url}} />
      <Card.Content style={styles.content}>
        <Title style={styles.titleStyle} numberOfLines={2}>
          {title}
        </Title>
        <Paragraph style={styles.descriptionStyle} numberOfLines={3}>
          {description}
        </Paragraph>
      </Card.Content>
      <Card.Actions style={styles.actions}>
        <TouchableOpacity onPress={onPress}>
          <Icon name="arrow-right-alt" size={25} style={{color: '#FFFF'}} />
        </TouchableOpacity>
      </Card.Actions>
    </Card>
  );
};

MainCard.propTypes = {
  description: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default MainCard;
