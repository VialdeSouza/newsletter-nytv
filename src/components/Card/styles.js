import {StyleSheet} from 'react-native';
import {useTheme} from 'react-native-paper';

const styles = () => {
  const {fonts, colors} = useTheme();

  return StyleSheet.create({
    item: {
      justifyContent: 'center',
      elevation: 0,
      backgroundColor: colors.primary,
    },
    description: {
      fontFamily: fonts.regular,
      color: colors.backdrop,
      textAlign: 'justify',
    },
    title: {
      fontFamily: fonts.medium,
      color: colors.accent,
    },
    actions: {
      justifyContent: 'flex-end',
    },
  });
};
export default styles;
