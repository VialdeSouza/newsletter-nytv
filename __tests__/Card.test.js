import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import Card from '../src/components/Card';

const url =
  'https://images.unsplash.com/photo-1612831455359-970e23a1e4e9?ixid=MXwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=625&q=80';

describe('Card', () => {
  it('should be render image cover', () => {
    const tree = renderer
      .create(<Card description="" title="" url={url} onPress={() => {}} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should be render with props', () => {
    const {getByText} = render(
      <Card
        url={url}
        onPress={() => {}}
        title="title test"
        description="description test"
      />,
    );
    expect(getByText('title test')).not.toBeNull();
    expect(getByText('description test')).not.toBeNull();
  });

  it('should be call function', () => {
    const action = jest.fn();

    const {getByTestId} = render(
      <Card
        url={url}
        onPress={action}
        title="title test"
        description="description test"
      />,
    );
    fireEvent.press(getByTestId('button'));
    expect(action).toHaveBeenCalled();
  });
});
