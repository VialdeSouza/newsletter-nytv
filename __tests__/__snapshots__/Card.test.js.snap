// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should be render image cover 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#6200ee",
      "borderRadius": 4,
      "elevation": 0,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#eeeeee",
            "height": 195,
            "overflow": "hidden",
          },
          Object {
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          undefined,
        ]
      }
    >
      <Image
        siblings={
          Array [
            "withTheme(Card.Cover)",
            "Card.Content",
            "Card.Actions",
          ]
        }
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1612831455359-970e23a1e4e9?ixid=MXwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=625&q=80",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "height": undefined,
              "justifyContent": "flex-end",
              "padding": 16,
              "resizeMode": "cover",
              "width": undefined,
            },
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          undefined,
          Object {
            "marginTop": 5,
          },
        ]
      }
    >
      <Text
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "500",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 20,
                  "letterSpacing": 0.15,
                  "lineHeight": 30,
                  "marginVertical": 2,
                },
                Object {
                  "color": "#03dac4",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "lineHeight": 25,
                },
              ],
            ],
          ]
        }
      >
        
      </Text>
      <Text
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.5)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "justify",
                },
              ],
            ],
          ]
        }
      >
        
      </Text>
    </View>
    <View
      index={2}
      siblings={
        Array [
          "withTheme(Card.Cover)",
          "Card.Content",
          "Card.Actions",
        ]
      }
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
      total={3}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="button"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 25,
              },
              Object {
                "color": "#FFFF",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          Ó•Å
        </Text>
      </View>
    </View>
  </View>
</View>
`;
